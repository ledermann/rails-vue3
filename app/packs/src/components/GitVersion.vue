<template>
  <div class="text-base">
    Version
    <code>{{ commitSha.substring(0, 7) }}</code
    >, created
    <b><time ref="timeElement" :datetime="commitTime"></time></b>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, onMounted, onBeforeUnmount } from 'vue';
import * as timeago from 'timeago.js';

export default defineComponent({
  name: 'GitVersion',

  props: {
    commitSha: {
      type: String,
      required: true,
    },

    commitTime: {
      type: String,
      required: true,
    },
  },

  setup() {
    const timeElement = ref();

    onMounted(() => {
      timeago.render(timeElement.value);
    });

    onBeforeUnmount(() => {
      timeago.cancel();
    });

    return {
      timeElement,
    };
  },
});
</script>
