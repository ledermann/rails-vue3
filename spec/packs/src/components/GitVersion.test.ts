import { mount } from '@vue/test-utils';
import GitVersion from '@/components/GitVersion.vue';

describe('GitVersion', () => {
  test('matches snapshot', () => {
    const wrapper = mount(GitVersion, {
      props: {
        commitSha: '1234567',
        commitTime: '2021-06-01T12:00:00+02:00',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();

    wrapper.unmount();
  });
});
