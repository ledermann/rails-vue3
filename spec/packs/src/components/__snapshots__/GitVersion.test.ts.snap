// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitVersion matches snapshot 1`] = `<div class="text-base"> Version <code>1234567</code>, created <b><time datetime="2021-06-01T12:00:00+02:00" timeago-id="3"></time></b></div>`;
